actioncable/lib/action_cable/server/worker.rb

```diff
@@ -1,39 +1,70 @@
-require 'celluloid'
 require 'active_support/callbacks'
+require 'concurrent'
 
 module ActionCable
   module Server
     # Worker used by Server.send_async to do connection work in threads. Only for internal use.
     class Worker
       include ActiveSupport::Callbacks
-      include Celluloid
 
-      attr_reader :connection
       define_callbacks :work
       include ActiveRecordConnectionManagement
 
+      def initialize(max_size=5)
+        @pool = Concurrent::ThreadPoolExecutor.new(
+          min_threads: 1,
+          max_threads: max_size,
+          max_queue: 0,
+        )
+      end
+
+      def connection
+        Thread.current[:connection] || raise("No connection set")
```

Comment:
I guess this should probably use [Module#thread_mattr_reader](https://github.com/rails/rails/blob/fdf359d8eb35de692bcf294f4cf323267d5e4cf0/activesupport/lib/active_support/core_ext/module/attribute_accessors_per_thread.rb)